setup.py

```diff
@@ -74,6 +74,10 @@
         's3fs',
         'tornado >= 5.1.1',
         'aiobotocore==0.10.0',
+        'aiohttp==3.3.2',
+        'botocore==1.12.49',
+        'boto3==1.9.49',
+        'awscli==1.16.59',
```

Comment:
The only gross thing about this is that when trying to deploy this, when we have _other_ packages with strongly pinned versions inside an environment (docker container, etc.), it's hard to resolve all of them. Our deployment strategy has been to install them in the order they allow us to be flexible with the `botocore` version. 😭

Reply:
@cHYzZQo (Peter), one of the leads of the image we build with recommends:

>  i would suggest not pinning
> or if you pin, be flexible about the patch release version

He can share more context about the awkwardness around botocore.

Reply:
At the very least, we should pin to the minor version and allow flexibility for the patch release.

Reply:
Let's try this instead. Let's keep the original code not the proposed changes. I will be unable to build autodoc on RTD using pip but that's a small price to pay vs maintainability.

I'm going to try switching the docs build to using conda to build the environment on RTD.

Reply:
Thank you and sorry about the difficulty around this one.

Reply:
No worries. There are some ugly dependencies around botocore and boto3 which are a bit surprising i.e. worst that I have seen in a while. I would be curious why there is awkwardness around botocore since it is also mentioned as challenging in the aiohttp source.

I have several workarounds in mind.

Reply:
They've started doing automated builds that trigger every couple days, constantly bumping the version even when there are no code changes. So any pinning you do very quickly falls behind 'latest' leading to all sorts of conflicts between other packages that depend on them.

Reply:
Thanks @cHYzZQo. Yeah, I noticed that their release cadence was way more frequent than most Python projects. aiobotocore pins these and that's part of the problem with dependency conflicts. Which makes me think that perhaps we float aiobotocore upwards `>=0.10.0` and not `pin ==` as well. Thoughts?

Reply:
yeah that would be fine i think